pmoves/services/evoswarm/persona_optimizer.py

```
@@ -473,33 +491,252 @@ async def _run_evoswarm_optimization(
+
+    def _pso_step(
+        self,
+        particle: _Particle,
+        global_best: Dict[str, Any],
+        rng: random.Random,
+        *,
+        inertia: float,
+        cognitive: float,
+        social: float,
+    ) -> None:
+        for field in self._numeric_fields():
```

Ruff (0.15.14)

[error] Import `field` from line 30 shadowed by loop variable

(F402)